cytominer_eval/operations/util.py

```diff
@@ -131,3 +133,87 @@ def get_grit_entry(df: pd.DataFrame, col: str) -> str:
         len(entries.unique()) == 1
     ), "grit is calculated for each perturbation independently"
     return str(list(entries)[0])
+
+
+class DistributionEstimator:
+    def __init__(self, arr):
+        self.sigma = EmpiricalCovariance().fit(arr)
+        
+    def mahalanobis(self, X):
+        return(self.sigma.mahalanobis(X))
+
+
+def calculate_mahalanobis(
+    pert_df: pd.DataFrame, control_df: pd.DataFrame
+    ) -> pd.Series:
+    """
+    Usage: Designed to be called within a pandas.DataFrame().groupby().apply()
+    """
+    assert len(control_df) > 1, "Error! No control perturbations found."
+        
+    # Get dispersion and center estimators for the 
+    control_estimators = DistributionEstimator(control_df)
+    
+    # Distance between mean of perturbation and control
+    maha = control_estimators.mahalanobis(
+                np.array(np.mean(pert_df, 0)).reshape(1, -1))[0]
+    return maha
+
+
+def default_mp_value_parameters():    
+    params = {"rescale_pca": True,
+              "nb_permutations": 100}
+    return(params)
+    
+    
+def calculate_mp_value(
+    pert_df: pd.DataFrame, control_df: pd.DataFrame,
+    params = {}
+    ) -> pd.Series:
+    """
+    Usage: Designed to be called within a pandas.DataFrame().groupby().apply()
+    """
+    assert len(control_df) > 1, "Error! No control perturbations found."
+    
+    # Assign parameters
+    p = default_mp_value_parameters()
+    assert (
+        all([x in p.keys() for x in params.keys()])
+    ), "Unknown parameters provided." 
```

Comment:
```suggestion
    ), "Unknown parameters provided. Only {e} are supported.".format(e=params.keys()) 
```

I didn't test this out, but doing this will make an assertion error more helpful and less frustrating to a user